credentials/generate-revocation-set.py

```diff
@@ -272,57 +350,68 @@ def get_revocation_points(self) -> list[dict]:
 
-    def get_issuer_cert(self, cert: x509.Certificate) -> Optional[x509.Certificate]:
+    def get_revocations_points_by_skid(self, issuer_subject_key_id) -> list[dict]:
         '''
-        Get the issuer certificate for
+        Get revocation points by subject key ID
 
         Parameters
         ----------
-        cert: x509.Certificate
-            Certificate
+        issuer_subject_key_id: str
+            Subject key ID
 
         Returns
         -------
-        str
-            Issuer certificate in PEM format
+        list[dict]
+            List of revocation points
         '''
-        issuer_name_b64 = get_issuer_b64(cert)
-        akid = get_akid(cert)
-        if akid is None:
-            return
 
-        # Convert CRL Signer AKID to colon separated hex
-        akid_hex = akid.hex().upper()
-        akid_hex = ':'.join([akid_hex[i:i+2] for i in range(0, len(akid_hex), 2)])
+        response = self.get_dcld_cmd_output_json(['query', 'pki', 'revocation-points',
+                                                  '--issuer-subject-key-id', issuer_subject_key_id])
+        logging.debug(f"Response revocation points: {response}")
+        return response["pkiRevocationDistributionPointsByIssuerSubjectKeyID"]["points"]
 
+    def get_certificate(self, issuer_name_b64: str, akid_hex: str) -> tuple[bool, Optional[x509.Certificate]]:
+        '''
+        Get certificate from DCL
+        '''
+        query_cmd_list = ['query', 'pki', 'x509-cert', '-u', issuer_name_b64, '-k', akid_hex]
         logging.debug(
-            f"Fetching issuer from:{self.rest_node_url}/dcl/pki/certificates/{issuer_name_b64}/{akid_hex}")
+            f"Fetching issuer from dcl query{' '.join(query_cmd_list)}")
+        response = self.get_dcld_cmd_output_json(query_cmd_list)
+        return self.get_first_approved_certificate(response, akid_hex)
 
-        if self.use_rest:
-            response = requests.get(
-                f"{self.rest_node_url}/dcl/pki/certificates/{issuer_name_b64}/{akid_hex}").json()
-        else:
-            response = self.get_dcld_cmd_output_json(
-                ['query', 'pki', 'x509-cert', '-u', issuer_name_b64, '-k', akid_hex])
 
-        issuer_certificate = response["approvedCertificates"]["certs"][0]["pemCert"]
+class RESTDCLDClient(DCLDClientInterface):
+    '''
+    A client for interacting with DCLD using the REST API.
 
-        logging.debug(f"issuer: {issuer_certificate}")
+    '''
 
-        try:
-            issuer_certificate_object = x509.load_pem_x509_certificate(bytes(issuer_certificate, 'utf-8'))
-        except Exception:
-            logging.error('Failed to parse PAA certificate')
-            return
+    def __init__(self, rest_node_url: str):
+        '''
+        Initialize the client
+
+        rest_node_url: str
+            RESTful API URL
+        '''
 
-        return issuer_certificate_object
+        self.rest_node_url = rest_node_url
```

Comment:
Can we add basic URL checks to make sure it's of the correct form (i.e. <protocol>://<uri>:<port>)?

Reply:
Yes adding checks.

Reply:
What are the allowed values preceding \.dcl? Also, do you need to worry about the trailing slash after \.org? Technically, one can have ":<port>" as optional HTTP URL, not sure if we should account for that with RegEx.

Reply:
It should just be the main or test net URLs: 
PRODUCTION_NODE_URL_REST = "https://on.dcl.csa-iot.org"
TEST_NODE_URL_REST = "https://on.test-net.dcl.csa-iot.org"

I've updated to limit the scope to just those subdomains and allow for an optional trailing slash. For the ":" i dont think i understand. The : in "https://" is not optional. I'll reach directly for clarification. Thanks

Reply:
I mean the port section which follows the FQDN, just to be comprehensive wih the pattern matching. For example, you can have http[s]://fqdn[:port].

Reply:
Of course, I see.  I've updated the regex and simplified it a bit to cover all these options.

Reply:
SG. Thanks.